### ISSUE TITLE
ShowTypingMiddleware should have better error handling

### ISSUE THREAD

**Issue by orsharab-zz** [microsoft/botbuilder-js#2527]
## Versions
NodeJS v10.15.3
BotBuilder NodeJS SDK v4.9.2
My app is written is typescript

## Describe the bug
ShowTypingMiddleware errors cannot be handled by onTurnError since it's running inside a setTimeout

## To Reproduce
Every error (usually socket hang up) generated from "await context.adapter.sendActivities" will cause an unhandled exception

## Expected behavior
Better error handling, maybe pass an error handler to the ShowTypingMiddleware constructor

## Additional context
Our app restarts on every unhandled exception so we can't use this middleware until this is fixed

**Comment by EricDahlvang** [comment]
@joshgummersall and @stevengum unfortunately, the exceptions in ShowTypingMiddleware continues to be an issue under some circumstances.  Maybe we should add an 'ignoreErrors' property to this middleware?

@orsharab please comment regarding the issues you are still running into here.

**Comment by orsharab** [comment]
BotBuilder NodeJS SDK v4.11.0
Still getting the following error: "Cannot perform 'get' on a proxy that has been revoked", I think the root cause it still the setTimeout that causes sometimes to loss context. I think it's best if we'll have the ability to ignore errors like @EricDahlvang suggested **before** trying to access TurnContext object on errors.

**Comment by joshgummersall** [comment]
This scenario could happen due to not clearing timeouts after an error occurs during middleware execution. Unfortunately, [this fix](https://github.com/microsoft/botbuilder-js/commit/d6088e2a67b221401ee63b9e4fd023e8de146501#diff-5d26fcf0cbc7bef36795039fd4d1d367e9ed853fed239085f6bfa0d7a1ff314c) did not make it into the `4.11` release stage. I've submitted a patch of this fix to the `4.11` branch, but it is unlikely that we will roll this out before the holiday season.

In the meantime, I would suggest adding a `try/catch` to your `adapter.onTurnError` handler. That should allow you to ignore these errors and ensure your bot does not restart unnecessarily.

**Comment by orsharab** [comment]
@joshgummersall this won't help since the error we're seeing is happening when trying to access the context object after the first error is caught.

**Comment by joshgummersall** [comment]
Right - but this would have the same effect as an `ignoreErrors` setting:

```typescript
adapter.onTurnError = (context, error) => {
  try {
    ...
  } catch (err) {
    console.error(err);
  }
}
```

Would that work for now?

**Comment by orsharab** [comment]
I don't see how this will help, from my understanding and from what I'm seeing in our logs, we're not reaching the onTurnError function.

**Comment by joshgummersall** [comment]
Apologies, I misunderstood the error scenario. This behavior is consistent with what I described [above](https://github.com/microsoft/botbuilder-js/issues/2527#issuecomment-735921397) and, as such, will be resolved once that change is published in 4.12.

**Comment by a-b-r-o-w-n** [comment]
Addressed in #2786. We expect to release 4.12 at the end of February, meanwhile you may access a daily build via npm.
